wp/lib/bap_wp/src/precondition.ml

```diff
@@ -321,6 +308,37 @@ let increment_stack_ptr (post : Constr.t) (env : Env.t) : Constr.t * Env.t =
   else
     post, env
 
+let lookup_precond (tid: Bap.Std.Tid.t) (env: Env.t) (post: Constr.t) =
+  match Env.get_precondition env tid with
+  | Some pre -> pre
+  | None ->
+    info "Precondition for return %s not found!" (Tid.to_string tid);
+    post
+
+let lookup_sub_handler (tid: Bap.Std.Tid.t) (env: Env.t) (post: Constr.t) =
+  match Env.get_sub_handler env tid with
+  | Some (Summary compute_func) -> compute_func env post tid
+  | Some Inline -> !inline_func post env tid
+  | None -> post, env
+
+let visit_call (call: Bap.Std.Call.t) (post : Constr.t) (env : Env.t) : Constr.t * Env.t =
+  let target = Call.target call in
+  let return = Call.return call in
+  match target, return with
+  | Direct _, Some (Indirect _)
+  | Indirect _, Some (Indirect _) -> post, env
+  | Indirect t_exp, None ->
+    Env.get_indirect_handler env t_exp env post t_exp false
+  | Direct t_tid, None ->
+    lookup_sub_handler t_tid env post
+  | Direct t_tid, Some (Direct r_tid) ->
+    let ret_pre = lookup_precond r_tid env post in
+    lookup_sub_handler t_tid env ret_pre
+  | Indirect t_exp, Some (Direct r_tid) ->
+    let ret_pre = lookup_precond r_tid env post in
+    Env.get_indirect_handler env t_exp env ret_pre t_exp true
```

Comment:
Can we add a log that we are visiting an indirect call for this case and the indirect case above?